### DISCUSSION TITLE
Rendering Error (GraphqlResponseError)

### DISCUSSION THREAD

**Discussion by Albermonte** [lowlighter/metrics#562] [Q&A] [answered]
Hi, I'm having some errors on the setup. No idea why :/

I copied my config from https://metrics.lecoq.io/ 
[This is my config](https://github.com/Albermonte/Albermonte/blob/006e6a4557755302f558b6814f59abc9e5109a86/.github/workflows/main.yml)

Error:
```
metrics/compute/albermonte/base > account user > failed : GraphqlResponseError: Request failed due to following response errors:
   - Something went wrong while executing your query. Please include `07C3:738B:1683D2:3072D5:6160840E` when reporting this issue.
```

[Full Log](https://pipelines.actions.githubusercontent.com/UKTHgOcc1CupN5ErcnSWBsZrZliuMQxWRW4vMogHzYmuu8IGZq/_apis/pipelines/1/runs/1/signedlogcontent/2?urlExpires=2021-10-08T18%3A10%3A36.7848224Z&urlSigningMethod=HMACV1&urlSignature=eKp69QfJoGJ%2FFH9CPqEC08xaZUmIdv%2Fn91GQHexl3js%3D)

**Comment by lowlighter** [comment, marked as answer]
Hi,
This is usually due to GitHub API timeouts when too much data is fetched at once 😕 

I tested your config and it should be fine. 
You could try using `repositories_batch: 5` (to only fetch 5 by 5 repos stats, or even a lower number) to avoid timeouts but I cannot guarantee it'll work.

Also I don't know which scopes you passed, but if it includes `repo` scope, it may also explain why it would work on the website/from my side but not on yours if you have a lot of private repositories. Unfortunately, users with a really large amount of repositories are not well supported because of GitHub GraphQL limitations

**Reply by Albermonte** [reply]
I only have 43 repos (is this too much?) and already tried with `repositories_batch: 5` and also `repositories_batch: 3`. Nothing works :/

**Reply by lowlighter** [reply]
No, 43 repositories should be fine

I'm not sure why you get the error though because when I tested your config I encountered no errors with Github Actions, but indeed I can clearly see that it doesn't work on your repo

I'll check again later to investigate, sorry you have trouble with metrics currently 😕

**Reply by lowlighter** [reply]
Ok so unfortunately I didn't found the cause of your error 😕 

Could you tell me which scopes you used for your personal access token, and also try to paste the following query in [GitHub GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer) and check the result?
<details>
<summary>
GraphQL Query
</summary>

```graphql
query BaseUser {
  user(login: "albermonte") {
    databaseId
    name
    login
    location
    createdAt
    avatarUrl
    websiteUrl
    isHireable
    twitterUsername
    repositories(last: 0 , isFork: false , ownerAffiliations: [OWNER]) {
      totalCount
      totalDiskUsage
      nodes {
        name
      }
    }
    packages {
      totalCount
    }
    starredRepositories {
      totalCount
    }
    watching {
      totalCount
    }
    sponsorshipsAsSponsor {
      totalCount
    }
    sponsorshipsAsMaintainer {
      totalCount
    }
    contributionsCollection {
      totalRepositoriesWithContributedCommits
      totalCommitContributions
      restrictedContributionsCount
      totalIssueContributions
      totalPullRequestContributions
      totalPullRequestReviewContributions
    }
    calendar:contributionsCollection(from: "2021-09-30T01:16:02.143Z", to: "2021-10-14T01:16:02.143Z") {
      contributionCalendar {
        weeks {
          contributionDays {
            color
          }
        }
      }
    }
    repositoriesContributedTo(includeUserRepositories: true) {
      totalCount
    }
    followers {
      totalCount
    }
    following {
    totalCount
    }
    issueComments {
      totalCount
    }
    organizations {
      totalCount
    }
  }
}
```

</details>

If you also get a `GraphqlResponseError` response error within the explorer, I would suggest to contact GitHub support and give them the query id.

Else you could also create a bot account and use its token in your workflow (seems weird, but it should work since [metrics.lecoq.io](https://metrics.lecoq.io) use a bot account, and I also tested it with my personal account too and both seems to be fine)

**Reply by Albermonte** [reply]
I tried with different scopes and none worked, right now it has `repo` scope and `read:user`

I already tried that query before opening this issue, read it on another one. All working, no errors.

**Reply by lowlighter** [reply]
Hello again 🙂 
Your issue may be resolved once #584 lands 
Actually, seems that any "huge stats" (not necessarly repositories) may cause a timeout.

Thanks for your patience 🙇

**Reply by Albermonte** [reply]
Awesome, I'll test it and let you know.

Thanks ❤️

**Reply by Albermonte** [reply]
As expected it runs successfully, but shows some errors. At least now we know where.

Private organizations were the issue, now with `read:org` seems to work. Using the public instance the number of organizations were 0 but using Github Actions it failed, no idea why...

~~Also it's still showing NaN but at least works.~~

Edit: now everything is working, seems like it was cached or something.

**Reply by lowlighter** [reply]
> Private organizations were the issue, now with read:org seems to work. Using the public instance the number of organizations were 0 but using Github Actions it failed, no idea why...

This is because your memberships are not public. If you open your profile in private browsing mode, you'll see that you're not part of any organizations ([see here for more infos](https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/organizations/memberships.md)). As the public instance use a bot account, it doesn't have access to "personal" data like a personal token would have

> Also it's still showing NaN but at least works.

You may still encounter this from time to time. From what I read, the GitHub GraphQL API timeout after 10 secs, so depending on your luck it may be able to compute the stat in time or not. Unfortunately I don't think that we can further improve this, it's more a GitHub issue now

Thanks a lot for your feedback 👍
